lib/linguist/languages.yml

```diff
@@ -3429,6 +3429,18 @@ Nemerle:
   tm_scope: source.nemerle
   ace_mode: text
   language_id: 243
+NEO:
+  type: programming
+  extensions:
+  - ".neo"
```

Comment:
Please update the search query you provided for this file extension. Right now, the results I checked where binary files and XML files. It's unclear how many, if any, are NEO.